### PR TITLE
Improving Developer Experience: Making Cypress More Developer Friendly

### PR DESCRIPTION
**Pull request by Ankit-Matth**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary

This pull request addresses the issue of missing auto-suggestion functionality and inefficient test case execution in Cypress. The implemented changes now provide auto-suggestion while writing test cases, improving the developer experience. Additionally, the test case execution process has been optimized to prevent automatic execution of all test cases upon minor changes, leading to a more streamlined workflow.

## Related Issue

This PR fixes the issue https://github.com/I-TECH-UW/OpenELIS-Global-2/issues/971

### Before:

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/146843890/da57a833-0d53-47a2-8be0-cfdfffc9aee6

### After:

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/146843890/6e54aee0-bd83-4e70-931b-e59595480225

### CODE REVIEW

**Comment by Ankit-Matth** [comment]
Hey @mozzy11, please review & merge this pull request.

File: frontend/cypress/pages/LoginPage.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/// <reference types="cypress" />
```
Comment (sherrif10) [inline frontend/cypress/pages/LoginPage.js:1]: is this neccessary ?

Comment (Ankit-Matth) [reply frontend/cypress/pages/LoginPage.js:1]: Absolutely, it enhances the developer experience by providing auto-suggestions while writing test cases.

Comment (CalebSLane) [reply frontend/cypress/pages/LoginPage.js:1]: hi @Ankit-Matth, possibly we have different extensions/versions in our vscode, but for me I can already get cypress suggestions while I type. When you set this option does this get applied globally, or just for the LoginPage.js?

**Pull request closed without merging**